kanidmd/src/lib/filter.rs

```diff
@@ -8,25 +8,28 @@
 //! [`Filter`]: struct.Filter.html
 //! [`Entry`]: ../entry/struct.Entry.html
 
-use crate::be::{IdxKey, IdxKeyRef, IdxKeyToRef, IdxMeta};
+use crate::be::{IdxKey, IdxKeyRef, IdxKeyToRef, IdxMeta, IdxSlope};
 use crate::identity::IdentityId;
 use crate::ldap::ldap_attr_filter_map;
 use crate::prelude::*;
 use crate::schema::SchemaTransaction;
 use crate::value::{IndexType, PartialValue};
-use hashbrown::HashSet;
+use concread::arcache::ARCacheReadTxn;
 use kanidm_proto::v1::Filter as ProtoFilter;
 use kanidm_proto::v1::{OperationError, SchemaError};
 use ldap3_server::proto::{LdapFilter, LdapSubstringFilter};
-// use smartstring::alias::String;
-use concread::arcache::ARCacheReadTxn;
-use smartstring::alias::String as AttrString;
+// use smartstring::alias::String as AttrString;
 use std::cmp::{Ordering, PartialOrd};
 use std::collections::BTreeSet;
 use std::hash::Hash;
 use std::iter;
+use std::num::NonZeroU8;
 use uuid::Uuid;
 
+use hashbrown::HashMap;
```

Comment:
How come we're using `hashbrown` when the `std` currently has the same implementation?

Reply:
Because it's not actually quite the same :) hashbrown uses ahash as the default hasher which is significantly faster than rust's std default hasher which is siphash. There are also some extra optimisations in hashbrown that aren't in the std version iirc.